src/DwellingImporter.php

```
@@ -0,0 +1,91 @@
+                                        );
+           $query->bindParam(':agentRef', $dwelling->AGENT_REF);
+           $query->bindParam(':address1', $dwelling->ADDRESS_1);
+           $query->bindParam(':address2', $dwelling->ADDRESS_2);
+           $query->bindParam(':town', $dwelling->TOWN);
+           $query->bindParam(':postcode', $dwelling->POSTCODE);
+           $query->bindParam(':description', $dwelling->DESCRIPTION);
+           $query->bindParam(':bedrooms', $dwelling->BEDROOMS);
+           $query->bindParam(':price', $dwelling->PRICE);
+           $query->bindParam(':image', $dwelling->IMAGE);
+           $query->bindParam(':typeId', $dwelling->TYPE);
+           $query->bindParam(':statusId', $dwelling->STATUS);
+           $query->execute();
+        }
+    }
+    /**
+     * inserts types from API
+     */
+    private function insertTypes(){
+        $types = $this->api->loadTypes();
+        foreach ($types as $type) {
+            $query = $this->db->prepare('INSERT INTO `types` (`type`) VALUES (:type)');
+            $query->bindParam(':type', $type->TYPE_NAME);
+            $query->execute();
+        }
+    }
+    /**
+     * inserts statuses from API
+     */
+    private function insertStatuses(){
+        $statuses = $this->api->loadStatuses();
+        foreach ($statuses as $status) {
+            $query = $this->db->prepare('INSERT INTO `statuses` (`status`) VALUES (:status)');
+            $query->bindParam(':status', $status->STATUS_NAME);
+            $query->execute();
+        }
+    }
+    /**
+     * deletes the data in the tables and inserts the data from the API
+     */
+    public function refreshDB(){
+        if ($this->api->checkIsUp())
+        {
+            $this->deleteTables();
+            $this->insertDwellings();
+            $this->insertTypes();
+            $this->insertStatuses();  
+        }
```

Some feedback if the api is down would be good, otherwise, your daemon will just fail silently.